File: WebPage/game_moves.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema
Imports System.Data.Entity.Spatial

Partial Public Class game_moves
    Public Property id As Integer

    Public Property gameId As Integer

    <Required>
    <StringLength(10)>
    Public Property whiteMove As String

    <StringLength(10)>
    Public Property blackMove As String

    Public Property moveOrder As Integer

    <StringLength(5)>
    Public Property annotation As String

    Public Overridable Property game As game
End Class

File: WebApplication/Models/game_moves.cs
namespace WebApplication
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;
    using System.Data.Entity.Spatial;

    public partial class game_moves
    {
        public int id { get; set; }

        public int gameId { get; set; }

        [Required]
        [StringLength(10)]
        public string whiteMove { get; set; }

        [StringLength(10)]
        public string blackMove { get; set; }

        public int moveOrder { get; set; }

        [StringLength(5)]
        public string annotation { get; set; }

        public virtual game game { get; set; }
    }
}
